docker_entrypoint.sh

```diff
@@ -40,189 +62,304 @@ ADV_TOR_MAXSOCKSCONNS=$(yq e '.advanced.tor.maxsocksconns' ${BITMONERO_DIR}/star
- echo "public-node=1" >> $new_conf_template
- if [ "$ADV_TOR_TORONLY" = "true" ] ; then
-  echo "# Advertise onion as public remote node (Communicated to wallet clients that crawl to our p2p network port, when public-node=1)" >> $new_conf_template
-  echo "anonymous-inbound=RPC_TOR_ADDRESS:MONERO_RPC_PORT_RESTRICTED,MONERO_ANON_INBOUND_HOST:MONERO_RPC_PORT_HS,ADV_TOR_MAXONIONCONNS" >> $new_conf_template
-  echo "# Disable UPnP port mapping" >> $new_conf_template
-  echo "igd=disabled" >> $new_conf_template
- #elif [ "ADV_P2P_UPNP" = "true" ] ; then
- # echo "#Enable UPnP port mapping" >> $new_conf_template
- # echo "igd=enabled" >> $new_conf_template
- fi
+ echo -e "\n# EXTERNAL CONNECTIONS" >> $config_file
+ echo "# Node advertisement: Requires --restricted-rpc, --rpc-bind-ip and --confirm-external-bind" >> $config_file
+ echo "# Advertise to wallets crawling the p2p network that they can use this node as a \"remote node\" for connecting their wallets." >> $config_file
+ echo "public-node=1" >> $config_file
 fi
 
-sed -i "s/TOR_HOSTNAME/$TOR_HOSTNAME/g" $new_conf_template
-sed -i "s/TOR_PORT/$TOR_PORT/g" $new_conf_template
-sed -i "s/ADV_TOR_MAXSOCKSCONNS/$ADV_TOR_MAXSOCKSCONNS/g" $new_conf_template
-sed -i "s/RPC_TOR_ADDRESS/$RPC_TOR_ADDRESS/g" $new_conf_template
-sed -i "s/MONERO_RPC_PORT_HS/$MONERO_RPC_PORT_HS/g" $new_conf_template
-sed -i "s/MONERO_RPC_PORT_RESTRICTED/$MONERO_RPC_PORT_RESTRICTED/g" $new_conf_template
-sed -i "s/MONERO_RPC_PORT/$MONERO_RPC_PORT/g" $new_conf_template
-sed -i "s/MONERO_ANON_INBOUND_HOST/$MONERO_ANON_INBOUND_HOST/g" $new_conf_template
-sed -i "s/ADV_TOR_MAXONIONCONNS/$ADV_TOR_MAXONIONCONNS/g" $new_conf_template
+sed -i "s|MONEROD_BIND_IP|$MONEROD_BIND_IP|g" $config_file
+sed -i "s|PEER_TOR_ADDRESS|$PEER_TOR_ADDRESS|g" $config_file
+sed -i "s|TOR_PORT|$TOR_PORT|g" $config_file
+sed -i "s|ADV_TOR_MAXSOCKSCONNS|$ADV_TOR_MAXSOCKSCONNS|g" $config_file
+sed -i "s|RPC_TOR_ADDRESS_RESTRICTED|$RPC_TOR_ADDRESS_RESTRICTED|g" $config_file
+sed -i "s|RPC_TOR_ADDRESS|$RPC_TOR_ADDRESS|g" $config_file
+sed -i "s|MONERO_P2P_PORT_LOCAL_BIND|$MONERO_P2P_PORT_LOCAL_BIND|g" $config_file
+sed -i "s|MONERO_P2P_PORT|$MONERO_P2P_PORT|g" $config_file
+sed -i "s|MONERO_RPC_PORT_RESTRICTED|$MONERO_RPC_PORT_RESTRICTED|g" $config_file
+sed -i "s|MONERO_RPC_PORT|$MONERO_RPC_PORT|g" $config_file
+sed -i "s|MONEROD_LOCAL_HOST|$MONEROD_LOCAL_HOST|g" $config_file
+sed -i "s|ADV_TOR_MAXONIONCONNS|$ADV_TOR_MAXONIONCONNS|g" $config_file
+sed -i "s|TOR_SOCKS_PROXY_HOST|$TOR_SOCKS_PROXY_HOST|g" $config_file
 
 #PRUNING config:
 if [ "$ADV_PRUNING_MODE" = "true" ] ; then
- echo -e "\n# PRUNING\nprune-blockchain=1" >> $new_conf_template
+ echo -e "\n# PRUNING\nprune-blockchain=1" >> $config_file
  #if [ "$ADV_PRUNING_SYNCPRUNEDBLOCKS" = "true" ] ; then
- # echo "sync-pruned-blocks=1" >> $new_conf_template
+ # echo "sync-pruned-blocks=1" >> $config_file
```

Comment:
```suggestion
 echo "sync-pruned-blocks=1" >> $config_file
```

Reply:
The interface for this isn't done.  Until it is, this feature will remain commented.  You're welcome to contribute the typescript that will enable it in the UI.  This is also conditional on whether the user turned it on.  I don't understand why you'd just turn it on for everybody.

Reply:
this is wrapped in an if statement that would only enable this if pruning was also enabled. This wouldn't enable the feature unless the node is pruned

Reply:
Yes but just because a user wants to prune their own blockchain doesn't mean they want to TRUST other nodes to do their pruning for them.  That would be another option, which should be highly discouraged in any case, and as such I didn't bother to write the interface for it.